### DISCUSSION TITLE
Better jest support

### DISCUSSION THREAD

**Discussion by IronLu233** [swc-project/swc#2048] [Ideas]
I try to use swc in my company project's unit test.

But I found there are some extra work for jest.

In general, we have to implement a `swc-jest` like [babel-jest](https://www.npmjs.com/package/babel-jest)

And I think the code for jest transform should not in @swc/core package.

How can I use a jest plugin, witch is build from Rust code?

**Comment by kdy1** [comment]
Currently, code for jest lives in `@swc/core`, because I'm still working on the rust plugin system.

For now, you can use `@swc/jest`, which depends on `@swc/core`.
https://www.npmjs.com/package/@swc/jest

**Reply by kdy1** [reply]
I'm working on rust-based plugin system, but it requires some decisions about APIs and it's delaying a lot.
I expect the plugin system itself not to be hard to implement, but determining API is very hard.

**Reply by IronLu233** [reply]
> you can use @swc/jest

I means, now `@swc/jest` is not work for some jest features.
For example, `jest.mock('module-path')`should hoist before module import.
and in jest test code can override modules named import see #2049 
Seems Rust plugin system is still in progress,
Is it means I have to implement "jest transform" using the Visitor API in https://swc.rs/docs/usage-plugin now?

**Reply by kdy1** [reply]
For jest, you can use https://github.com/swc-project/swc/tree/71080dbd26fdb09bc6b6e69b3e088e3994b704ce/ecmascript/ext-transforms

**Reply by IronLu233** [reply]
Yes, this is what I need. I will try to use it and make it better with some PRs.
